src/ComunicaEngine.js

```diff
@@ -77,23 +77,29 @@ export default class ComunicaEngine {
     let sources = await source;
     if (!sources)
       return null;
+
     // Strip the fragment of a URI
     if (sources instanceof URL)
       sources = sources.href;
     else if (sources.termType === 'NamedNode')
       sources = sources.value;
+
     if (typeof sources === 'string')
       sources = [sources.replace(/#.*/, '')];
-    // Await multiple promises in an array
+    // Recurse arrays
     else if (Array.isArray(sources))
-      sources = await Promise.all(sources);
+      sources = [].concat(...(await Promise.all(sources.map(src => this.toComunicaSources(src)))));
+    // Needs to be after the string check since those also have a match functions
+    else if (sources.match)
+      sources = [Object.assign(sources, { type: 'rdfjsSource' })];
```

Comment:
We probably don't want to alter `sources`. How about swapping the args to `assign`?